src/librustc_mir/transform/mod.rs

```diff
@@ -491,6 +475,27 @@ fn run_optimization_passes<'tcx>(
         &dump_mir::Marker("PreCodegen"),
     ];
 
+    // First we run the optimzations that are required for generator lowering, then we lower the
+    // generator aggregates to field assignments.
```

Comment:
I don't understand this comment.

Reply:
Seems like that would belong to `let optimizations_with_generators`? Also the main thing generator lowering does is constructing the state machine; that this also removes generator aggregates is a tiny detail.

Reply:
we don't just change the dialect by deaggregating `AggregateKind::Generator`, we also need to run some optimizations before doing that.

Reply:
Sure. But that is all controlled by the phase list above, not by the run_pass invocation down here.

Reply:
in that case I don't know what comments you wanted me to write on the `run_pass` invocation. I can just remove the comments, but explaining this boilerplate seems weird

Reply:
I meant to ask for comments around the `let` bindings that delllcare the lists of optimizations.

The boilerplate here probably just needs something pointing out that the MirPhase is different:
```rust
// Generator lowering
run_pass(...);
// Main optimization pass
run_pass(...);
```